cmd/kubelet/app/server.go

```diff
@@ -138,15 +138,16 @@ const (
 )
 
 // NewKubeletCommand creates a *cobra.Command object with default parameters
-func NewKubeletCommand() *cobra.Command {
+func NewKubeletCommand(ctx context.Context) *cobra.Command {
+	logger := klog.FromContext(ctx)
```

Comment:
is it OK to initialize logger before `logs.InitLogs()` was called? How would it be configured? How we will report an error in line 150?

Reply:
However, it's still better to retrieve the logger *after* logging configuration. I think it works (get logger, reconfigure, logger changes behavior), but if get the logger later, you get a different, slightly more performant logger (not that it matters much here).

Reply:
@pohly 
I appreciate your help!
If we get the logger after `logs.InitLogs()`, the logs at lines 150 and 216 won't support contextual logging. Do you think it makes sense to remove `contextual k8s.io/kubernetes/cmd/kubelet/.*` from `logcheck.conf` in this case? Would like to hear your thoughts

Reply:
I would keep it. Support for contextual logging is going to get as good as it gets. Sometimes logging something early simply cannot be avoided.

Reply:
Thank you, Pohly. 
I'll resolve this conversation then.

Reply:
Thank you for your suggestion.
I believe it is better👍
By retrieving the logger again after calling `logs.InitLogs()`, we can use a logger that correctly reflects the updated log settings. 
I will modify the code to call `klog.FromContext(ctx)` again after `logs.InitLogs()`. 
Here’s the updated code:
```
func NewKubeletCommand(ctx context.Context) *cobra.Command {
    // Retrieve the logger initially (with default settings)
    logger := klog.FromContext(ctx)
    // Initialize logs
    logs.InitLogs()
    // Retrieve the logger again to reflect the updated log settings
    logger = klog.FromContext(ctx)
}
```

Reply:
@SergeyKanzhelev @pohly 
Do you both think this makes sense?

Reply:
yes, please do this